include/clang/AST/Decl.h

```diff
@@ -800,6 +800,15 @@ class VarDecl : public DeclaratorDecl, public Redeclarable<VarDecl> {
   /// C++ default argument.
   mutable InitType Init;
 
+  // TODO: like the Init member above, it wastes space to have a pointer to a
+  // BoundsExpr in every VarDecl when many of them won't have bounds
+  // declarations. We could move the Init member and the BoundsExpr to an
+  // "extension" object that is allocated on demand, at least not increasing
+  // the space usage of every single VarDecl.
+
+  /// \brief The bounds expression for the variable, if any.
+  Expr *BoundsExpr;
```

Comment:
Should this be a BoundsExpr *?  Or are there cases where you'd need to hang some other kind of expression of it?